src/hooks/useMedia.ts

```diff
@@ -0,0 +1,64 @@
+import { useEffect, useState } from "react";
+
+type UseMediaType = {
+    cameraList: InputDeviceInfo[];
+    micList: InputDeviceInfo[];
+    setCamera: React.Dispatch<React.SetStateAction<InputDeviceInfo | undefined>>;
+    setMic: React.Dispatch<React.SetStateAction<InputDeviceInfo | undefined>>;
+    selectedCamera: InputDeviceInfo | undefined;
+    selectedMic: InputDeviceInfo | undefined;
+    mediaStream: MediaStream | null;
+}
+
+const useMedia: () => UseMediaType = () => {
+    const [cameraList, setCameraList] = useState<InputDeviceInfo[]>([]);
+    const [micList, setMicList] = useState<InputDeviceInfo[]>([]);
+    
+    const [selectedCamera, setCamera] = useState<InputDeviceInfo>();
+    const [selectedMic, setMic] = useState<InputDeviceInfo>();
+
+    const [mediaStream, setMediaStream] = useState<MediaStream | null>(null);
+
+    useEffect(() => {
+        getMediaDevicesList();
+        
+        navigator.mediaDevices.getUserMedia({audio:true,video:true})
```

Comment:
should run `getUserMedia` first, then enumrate the devices, it's already mentioned in the StackOverflow link in the issue